relay/relay.go

```diff
@@ -127,6 +128,7 @@ func NewRelay(config *Config, name string) (*Relay, error) {
 	relay.timer = NewDefaultTimer()
 	relay.log = logger.GetOrCreate(name)
 	relay.signatures = make(map[core.PeerID][]byte)
+	relay.elrondPublicAddress = hex.EncodeToString(elrondBridge.Address().AddressBytes())
```

Comment:
you can get a string out of easier with `AddressAsBech32String`

Reply:
Looking at the code I don't actually this the address needs to be stored it the data structure. As it's only needed one time when the relay joins it should be simply grabbed at that time [of use](https://github.com/ElrondNetwork/elrond-eth-bridge/blob/f358a6849a3b0a01df1ff8e3b012f77347a12aaf/relay/relay.go#L322)

Reply:
> you can get a string out of easier with `AddressAsBech32String` so this line will be
> 
> ```
> relay.elrondPublicAddress = elrondBridge.Address().AddressAsBech32String()
> ```

I have tried this before raising the CR but during the isWhitelisted check the elrondPublicAddress has to be the decoded value from bech32 without the erd tag.

```
ERROR[2021-09-21 12:13:42.031]   executeQuery: 'erd1r69gk66fmedhhcg24g2c5kn2f2a5k4kvpr6jfw67dn2lyydd8cfswy6ede' is not a valid hex string: encoding/hex: invalid byte: U+0072 'r' 
ERROR[2021-09-21 12:13:42.031]   A peer with address "erd1r69gk66fmedhhcg24g2c5kn2f2a5k4kvpr6jfw67dn2lyydd8cfswy6ede" tryed to join but is not whitelisted 
```

Reply:
> Looking at the code I don't actually this the address needs to be stored it the data structure. As it's only needed one time when the relay joins it should be simply grabbed at that time [of use](https://github.com/ElrondNetwork/elrond-eth-bridge/blob/f358a6849a3b0a01df1ff8e3b012f77347a12aaf/relay/relay.go#L322)
> 
> PS: this will also allow us to test cover this change.

You are right, we could get the address just for the flow where we do the broadcast.